### PR TITLE
SLING-11150: Return package id in the DistributionResponse for a dete…

### PR DESCRIPTION
**Pull request by amit-jain**
…rministic way of identifying status

- Add DistributionResponseInfo object to DistributionResponse

### CODE REVIEW

**Comment by amit-jain** [comment]
@tmaret Could you please review. Thanks!

File: src/main/java/org/apache/sling/distribution/DistributionResponse.java [hunk 1]
```diff
@@ -57,4 +57,13 @@
      */
     @Nullable
     String getMessage();
+
+    /**
+     * returns additional properties related to the {@link DistributionRequest}
+     * 
+     * @return additional properties
+     */
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/DistributionResponse.java:65]: If I understand the intention properly, this method should be annotated with `@Nonnull`

File: src/test/java/org/apache/sling/distribution/DistributionResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        @Nonnull @Override public DistributionRequestState getState() {
+            return state;
+        }
+
+        @Nullable @Override public String getMessage() {
+            return message;
+        }
+        
+        @Override public DistributionResponseInfo getDistributionInfo() {
```
Comment (tmaret) [inline src/test/java/org/apache/sling/distribution/DistributionResponseTest.java]: Should be annotated `@Nonnull`

**Review by tmaret — COMMENTED** [review]
The api module is using Java 8 so it's technically possible to use default methods. But I think we should limit our use of `default` for ConsumerType to avoid breaking consumers. In this case we provide all the implementations and would be fine without providing a default implementation in the interface.

**Review by tmaret — APPROVED** [review]
LGTM

**Pull request merged**